### PR TITLE
Fixed mkdir invalid argument error when registering workflow

### PR DESCRIPTION
**Pull request by CamelCaseCam**
I had a bug where workflow registration would fail while creating a log directory and log file because linux (or at least Ubuntu) doesn't allow colons in file names. This change fixes that by replacing the colons in the log path with underscores. I've attached the error logs and a screenshot of the error. 

image
[.latch_report.tar.gz](https://github.com/latchbio/latch/files/9847043/default.latch_report.tar.gz)

### CODE REVIEW

**Comment by CamelCaseCam** [comment]
@ayushkamat This is the one I was talking about. I've traced this and a similar bug preventing ssh credentials from being added to making the workflow on an NTFS file system running under Linux. NTFS doesn't allow colons in file names, and it doesn't support permissions, so there was no way to register workflows (both --remote and regular was broken)

File: latch_cli/services/register/register.py [hunk 1]
```diff
@@ -58,9 +58,10 @@ def _print_window(curr_lines: List[str], line: str):
 def _print_and_save_build_logs(build_logs, image: str, pkg_root: Path):
     print(f"Building Docker image for {image}")
 
-    os.makedirs(str(pkg_root) + f"/.logs/{image}", exist_ok=True)
+    # Not sure if I should change the value of `image` or not
+    os.makedirs(str(pkg_root) + f"/.logs/{image.replace(':', '_')}", exist_ok=True)
```
Comment (ayushkamat) [inline latch_cli/services/register/register.py:62]:
```suggestion
    logs_path = Path(pkg_root).joinpath(".logs").joinpath(image).resolve()
    logs_path.mkdir(exist_ok=True)
```

File: latch_cli/services/register/register.py [hunk 2]
```diff
@@ -58,9 +58,10 @@ def _print_window(curr_lines: List[str], line: str):
...
     with open(
-        str(pkg_root) + f"/.logs/{image}/docker-build-logs.txt", "w"
+        str(pkg_root) + f"/.logs/{image.replace(':', '_')}/docker-build-logs.txt", "w"
```
Comment (ayushkamat) [inline latch_cli/services/register/register.py:64]:
```suggestion
        logs_path.joinpath("docker-build-logs.txt"), "w"
```

**Review by ayushkamat — CHANGES_REQUESTED** [review]
heres a cleaner solution imo, let me know if you have any thoughts and if not ill merge

**Comment by CamelCaseCam** [comment]
@ayushkamat Looks great! (I've also got a local fix for --remote ssh not working, but it's a bit of a hack, so I'm not sure if it's ready to add)

**Pull request merged**